docs/control-structures.rst

```diff
@@ -427,28 +434,42 @@ Exceptions can be caught with the ``try``/``catch`` statement.
 The convenience functions ``assert`` and ``require`` can be used to check for conditions and throw an exception
 if the condition is not met.
 
-The ``assert`` function should only be used to test for internal errors, and to check invariants. Properly functioning code should never reach a failing ``assert`` statement; if this happens there is a bug in your contract which you should fix. Language analysis tools can evaluate your contract to identify the conditions and function calls which will reach a failing ``assert``.
+The ``assert`` function should only be used to test for internal
+errors, and to check invariants. Properly functioning code should
+never reach a failing ``assert`` statement; if this happens there
+is a bug in your contract which you should fix. Language analysis
+tools can evaluate your contract to identify the conditions and
+function calls which will reach a failing ``assert``.
 
 An ``assert``-style exception is generated in the following situations:
 
-#. If you access an array at a too large or negative index (i.e. ``x[i]`` where ``i >= x.length`` or ``i < 0``).
+#. If you access an array or array slice at a too large or negative index (i.e. ``x[i]`` where ``i >= x.length`` or ``i < 0``).
 #. If you access a fixed-length ``bytesN`` at a too large or negative index.
 #. If you divide or modulo by zero (e.g. ``5 / 0`` or ``23 % 0``).
 #. If you shift by a negative amount.
 #. If you convert a value too big or negative into an enum type.
 #. If you call a zero-initialized variable of internal function type.
 #. If you call ``assert`` with an argument that evaluates to false.
 
-The ``require`` function should be used to ensure valid conditions that cannot be detected until execution time.
-These conditions include inputs, or contract state variables are met, or to validate return values from calls to external contracts.
+The ``require`` function should be used to ensure valid conditions
+that cannot be detected until execution time.
+These conditions include inputs, or contract state variables are met,
```

Comment:
What does `or contract state variables are met` mean?

Reply:
Good question!

Reply:
Removed the state vars.